friendships.go

```diff
@@ -0,0 +1,241 @@
+	}
+	return &UserResponse{
+		User:      res,
+		RateLimit: getRateLimit(resp.Header),
+	}, nil
+}
+
+type friendshipsDetail struct {
+	RelationshipDetail `json:"relationship"`
+}
+
+// FriendshipsUpdate calls the Twitter /friendships/update.json endpoint.
+func (c *Client) FriendshipsUpdate(ctx context.Context, screenName,
+	userID string, device, retweets bool) (*RelationshipDetailResponse, error) {
+	values := url.Values{}
+	if screenName != "" {
+		values.Set("screen_name", screenName)
+	}
+	if userID != "" {
+		values.Set("user_id", userID)
+	}
+	values.Set("device", strconv.FormatBool(device))
+	values.Set("retweets", strconv.FormatBool(retweets))
+
+	resp, err := c.do(ctx, "POST",
+		"https://api.twitter.com/1.1/friendships/update.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res friendshipsDetail
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return nil, err
+	}
+	return &RelationshipDetailResponse{
+		Relationship: res.RelationshipDetail,
+		RateLimit:    getRateLimit(resp.Header),
+	}, nil
+}
+
+// FriendshipsShow calls the Twitter /friendships/show.json endpoint.
+func (c *Client) FriendshipsShow(ctx context.Context, sourceID, targetID int64,
+	sourceScreenName, targetScreenName string) (*RelationshipDetailResponse, error) {
+	values := url.Values{}
+	if sourceID > 0 {
+		values.Set("source_id", strconv.FormatInt(sourceID, 10))
+	}
+	if targetID > 0 {
+		values.Set("target_id", strconv.FormatInt(targetID, 10))
+	}
+	if sourceScreenName != "" {
+		values.Set("source_screen_name", sourceScreenName)
+	}
+	if targetScreenName != "" {
+		values.Set("target_screen_name", targetScreenName)
+	}
+	resp, err := c.do(ctx, "GET",
+		"https://api.twitter.com/1.1/friendships/show.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res friendshipsDetail
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return nil, err
+	}
+	return &RelationshipDetailResponse{
+		Relationship: res.RelationshipDetail,
+		RateLimit:    getRateLimit(resp.Header),
+	}, nil
+}
+
+type friendshipsRelationship struct {
+	Relationship `json:"relationship"`
+}
+
+// FriendshipsLookup calls the Twitter /friendships/lookup.json endpoint.
+func (c *Client) FriendshipsLookup(ctx context.Context,
+	screenNames, userIDs []string) (*RelationshipResponse, error) {
+	values := url.Values{}
+	if len(screenNames) > 0 {
+		values.Set("screen_name", strings.Join(screenNames, ","))
+	}
+	if len(userIDs) > 0 {
+		values.Set("user_id", strings.Join(userIDs, ","))
+	}
+	resp, err := c.do(ctx, "GET",
+		"https://api.twitter.com/1.1/friendships/lookup.json", values)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if err = checkResponse(resp); err != nil {
+		return nil, err
+	}
+	var res friendshipsRelationship
```

Comment:
From the endpoint description, it looks like this returns an array of `Relationship`s.